### PR TITLE
Fix MediaContent height when used inside a modal

### PR DESCRIPTION
**Pull request by mcellier**
Before:
Capture d’écran 2020-03-20 à 15 36 43

After:
Capture d’écran 2020-03-20 à 15 36 05

### CODE REVIEW

File: src/_utils/closest.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  let el = this
+
+  do {
+    if (el.matches(s)) return el
+    el = el.parentElement || el.parentNode
+  } while (el !== null && el.nodeType === 1)
+  return null
+}
+
+if (!Element.prototype.closest) {
```
Comment (mcellier) [inline src/_utils/closest.tsx]: Element is not defined server-side. Adding something to handle this

File: src/layout/content/content.tsx [hunk 2]
```diff
@@ -17,11 +18,17 @@ export class MainContent extends Component<MainContentProps> {
...
     }
+
     const topBar: HTMLElement = document.querySelector(this.props.topBarSelector)
-    const height = topBar
-      ? Math.floor(window.innerHeight - topBar.offsetHeight)
-      : window.innerHeight
+    let minus = 0
+
+    if (this.mainContentRef.current.closest('.kirk-modal--hasCloseButton')) {
+      minus = 70 // padding-top of modals with close button
```
Comment (gabrielagabriel) [inline src/layout/content/content.tsx]: **No magic numbers**: is that possible to create a variable for that?

File: src/layout/content/content.tsx [hunk 3]
```diff
@@ -17,11 +18,17 @@ export class MainContent extends Component<MainContentProps> {
...
+    } else if (topBar) {
+      minus = topBar.offsetHeight
+    }
```
Comment (gabrielagabriel) [inline src/layout/content/content.tsx]: else... it will receive `0`?

Comment (mcellier) [reply src/layout/content/content.tsx]: Yes. I've let the previous behavior in place (no topbar).

**Comment by gabrielagabriel** [comment]
IMHO these kind of height issue are fixed with **Flex-Box** does the _bottom height_ is a dynamic variable?

**Comment by mcellier** [comment]
> IMHO these kind of height issue are fixed with **Flex-Box** does the _bottom height_ is a dynamic variable?

I'll let you talk with @PerrineBoissieres who implemented this at first. She'll be better than me at explaining the context

File: src/layout/content/content.tsx (hunk 2 above)
Comment (mcellier) [reply src/layout/content/content.tsx]: Added the size into `_utils/banding`'s `componentSizes`

**Comment by PerrineBoissieres** [comment]
@gabrielagabriel `MainContent`, `Content` and `BottomContent` use flexbox positionning to have bottom-aligned elements in a page, and it handles mobile browser weird behaviors as well as window resizing :)
`MainContent` has a prop to take in account a specific component (like a topbar) for height calculations, but it doesn't work for modals as they use a top padding. This PR implements a workaround so the `MainContent` can ignore the padding for the close button.

**Pull request merged**